src/Progress/ProgressUpdateRepository.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace BristolSU\Support\Progress;
+
+use BristolSU\Support\Progress\Contracts\ProgressUpdateContract;
+use Illuminate\Support\Facades\Hash;
+
+class ProgressUpdateRepository implements ProgressUpdateContract {
```

Comment:
This repository should be responsible for checking if a progress has been updated. The reason for the repository is to let us swap how/where progress hashes are saved, therefore this should have an API to allow a progress update to be stored and a to check if a progress has been updated

See snapshot class for more details